tests/conftest.py

```diff
@@ -0,0 +1,21 @@
+import logging
+import os
+import sys
+
+import dotenv
+import pytest
+
+from dbtools.pg import Postgres, load_pgconfig
+
+
+dotenv.load_dotenv('.env.test')
+
+@pytest.fixture(name="pgconfig")
+def fixture_pgconfig():
+    return load_pgconfig()
+
+
+@pytest.fixture()
+def db_src(pgconfig) -> Postgres:
+    with Postgres(**pgconfig.non_wildcard_atts) as db:
+        yield db
```

Comment:
Most just curious, they seem interchangeable on first glance, but why `yield` over `return` here?

Reply:
That's what's recommended for pytest fixtures. You can basically put any test set up before the `yield` and any tear down after. In this case, while there's nothing after the `yield`, I think it I `return`'d the database connection it would exit the `with` block and class `Postgres`'s `__exit__()` method which closes the connection. So I used yield in the `with` to keep the connection alive.

Reply:
Interesting, thanks for the explainer. I guess I'm realizing that the "invisible" call to `__exit__()`, implicit to the use of `with`, can be thought of as the tear-down code in this case (?)

Reply:
Yep exactly